app/controllers/api/notes_controller.rb

```diff
@@ -294,11 +294,25 @@ def search
           raise OSM::APIBadUserInput, "Date #{params[:to]} is in a wrong format"
         end
 
-        @notes = @notes.where(:created_at => from..to)
+        @notes = params[:sort] == "created_at" ? @notes.where(:created_at => from..to) : @notes.where(:updated_at => from..to)
       end
 
       # Find the notes we want to return
-      @notes = @notes.order("updated_at DESC").limit(result_limit).preload(:comments)
+      @notes = if params[:sort] == "created_at"
+                 if params[:order] == "oldest"
+                   @notes.order("created_at ASC")
+                 else
+                   @notes.order("created_at DESC")
+                 end
+               else
+                 if params[:order] == "oldest"
+                   @notes.order("updated_at ASC")
+                 else
+                   @notes.order("updated_at DESC")
+                 end
+               end
+
+      @notes = @notes.distinct.limit(result_limit).preload(:comments)
```

Comment:
BTW: I also added `.distinct` which was missing before but might also be useful because returning the same note multiple times is not really helpful...

Reply:
This does fix a big problem with duplicate notes, see https://github.com/ENT8R/NotesReview/issues/43